File: php/sleep_in.php
<?php

	function sleep_in($is_weekday,$is_on_vacation)
	{
		return (!$is_weekday || $is_on_vacation);
	}
echo sleep_in(False, False) ? "pass" : "fail"; echo "\n";
echo sleep_in(True, False) ? "fail" : "pass"; echo "\n";
echo sleep_in(False, True) ? "pass" : "fail"; echo "\n";
?>

File: ruby/sleep_in.rb
def sleep_in(is_weekday,is_on_vacation)
	return !is_weekday || is_on_vacation
end
puts sleep_in(false, false) ? "pass" : "fail"
puts sleep_in(true, false) ? "fail" : "pass" 
puts sleep_in(false, true) ? "pass" : "fail"